src/sonic-yang-models/yang-models/sonic-port.yang

```diff
@@ -63,6 +63,44 @@ module sonic-port{
 					}
 				}
 
+				leaf autoneg {
+					description "Port auto negotiation mode";
+
+					type string {
+						pattern "on|off";
+					}
+				}
+				
+				leaf-list adv_speeds {
+					description "Port advertised speeds, valid value could be a list of interger or all";
+
+					type union {
+						type uint32 {
+							range 1..400000;
```

Comment:
So there is no other constraint on adv speed?, it can take any number of values. Kindly point to the correct paragraph in HLD for this plz, I will have a look. Thx

Reply:
Yes, there is no other constraint for adv_speeds, I am following the field "speed" in the same yang model file. HLD: https://github.com/Junchao-Mellanox/SONiC/blob/port-auto-neg/doc/port_auto_neg/port-auto-negotiation-design.md#config-advertised-speeds